src/index.js

```diff
@@ -10,34 +10,30 @@ import {
   useConstraints,
   useTranslator,
   useSelections,
-  useTheme,
   usePromise,
   useKeyboard,
   useRect,
   useApp,
 } from '@nebula.js/stardust';
 
-import useReactRoot from './table/context/useReactRoot';
-import registerLocale from './locale/src';
 import properties from './object-properties';
 import data from './data';
 import ext from './ext';
 import manageData from './handle-data';
-import sortingFactory from './sorting-factory';
 import { render, teardown } from './table/Root';
-import muiSetup from './mui-setup';
-import tableThemeColors from './table-theme-colors';
+import useReactRoot from './nebula-hooks/use-react-root';
+import useAnnounceAndTranslations from './nebula-hooks/use-announce-and-translations';
+import useSorting from './nebula-hooks/use-sorting';
+import useExtendedTheme from './nebula-hooks/use-extended-theme';
 
 const initialPageInfo = {
   page: 0,
   rowsPerPage: 100,
   rowsPerPageOptions: [10, 25, 100],
 };
 const nothing = async () => {};
-const renderWithCarbon = ({ env, translator, rootElement, model, theme, selectionsAPI, app, rect, layout }) => {
-  if (env.carbon) {
-    registerLocale(translator);
-    const changeSortOrder = sortingFactory(model);
+const renderWithCarbon = ({ env, rootElement, model, theme, selectionsAPI, app, rect, layout, changeSortOrder }) => {
+  if (env.carbon && changeSortOrder && theme) {
     render(rootElement, { layout, model, manageData, theme, selectionsAPI, changeSortOrder, app, rect });
```

Comment:
@vcellu  is the render from Root.jsx or Root.native.jsx?

Reply:
yes it is, we need to sync on how this works. Will setup a meeting, let's merge this for now